### PR TITLE
Config worker use public frontend endpoint

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Config worker use public frontend endpoint

**Why?**
Config worker use public frontend endpoint should use the endpoint from public client config.

**How did you test it?**

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

**Comment by dnr** [comment]
Why not use the internal endpoint?

**Comment by yux0** [comment]
> Why not use the internal endpoint?

It should. See: https://github.com/temporalio/temporal/pull/3160

**Review by meiliang86 — APPROVED** [review]

**Review by yycptt — APPROVED** [review]

**Comment by dnr** [comment]
> It should. See: #3160

I saw that, I'm just confused how these relate to each other

File: service/worker/fx.go [hunk 1]
```diff
@@ -89,6 +101,13 @@ func ConfigProvider(
 	)
 }
 
+func FrontendClientProvider(
```
Comment (hehaifengcn) [inline service/worker/fx.go:104]: can we rename to `LocalFrontendClientProvider` and also add some comment that why `PublicClient` is actually `Local endpoint`?

**Pull request merged**